File: device-sdk/src/main/java/io/oddworks/device/metric/OddVideoLoadMetric.java
package io.oddworks.device.metric;


public class OddVideoLoadMetric extends OddMetric {
    private static final String TAG = OddVideoLoadMetric.class.getSimpleName();

    protected static String videoLoadAction = ACTION_VIDEO_LOAD;
    protected static boolean videoLoadEnabled = false;

    public OddVideoLoadMetric() {

    }

    public static void setAction(String action) {
        videoLoadAction = action;
    }

    public static void setEnabled(boolean enabled) {
        videoLoadEnabled = enabled;
    }

    @Override
    public String getAction() {
        return videoLoadAction;
    }

    @Override
    public boolean getEnabled() {
        return videoLoadEnabled;
    }
}

File: device-sdk/src/main/java/io/oddworks/device/metric/OddVideoLoadMetric.kt
package io.oddworks.device.metric

import org.json.JSONObject


class OddVideoLoadMetric(contentType: String, contentId: String, title: String, meta: JSONObject? = null) : OddMetric(contentType, contentId, title, meta) {

    override val action: String
        get() = OddVideoLoadMetric.action

    override val enabled: Boolean
        get() = OddVideoLoadMetric.enabled

    companion object {
        private val TAG = OddVideoLoadMetric::class.java.simpleName

        var action = OddMetric.ACTION_VIDEO_LOAD
        var enabled = false
    }
}
